install/0000_90_cluster-version-operator_02_servicemonitor.yaml

```diff
@@ -9,9 +9,13 @@ metadata:
     exclude.release.openshift.io/internal-openshift-hosted: "true"
 spec:
   endpoints:
+  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
```

Comment:
Are there API docs for ServiceMonitor?  [OpenShift docs][1] link [Prometheus docs][2], but I don't see anyone talking about `bearerTokenFile`.  Ah, [here we go][3].  But I don't see anything there about why we'd expect this path to work, who's putting what token in that path, or who's consuming it.  Would be good to get metrics input on this.

[1]: https://docs.openshift.com/container-platform/4.3/operators/operator_sdk/osdk-monitoring-prometheus.html#osdk-monitoring-prometheus-servicemonitor_osdk-monitoring-prometheus
[2]: https://github.com/coreos/prometheus-operator/blob/7a25bf6b6bb2347dacb235659b73bc210117acc7/Documentation/design.md#servicemonitor
[3]: https://github.com/coreos/prometheus-operator/blob/0c6d184c19c890af10d715310d1770424d4e370b/Documentation/api.md#apiserverconfig

Reply:
the token file is automatically made available by the [Admission Controller](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#service-account-admission-controller).

Reply:
Also there are 2 endpoint definitions. Maybe you wanted this rather:

```yaml
- bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
  interval: 30s
  port: https
  scheme: https
  tlsConfig:
    caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
    serverName: metrics.openshift-cluster-version.svc
```